forgerock-auth/src/test/java/org/forgerock/android/auth/ServerConfigTest.java

```diff
@@ -230,7 +233,102 @@ public void onResponse(@NotNull Call call, @NotNull Response response) throws IO
 
         });
         countDownLatch.await();
-        Assert.assertFalse(result[0]);
+        assertFalse(result[0]);
+    }
 
+    @Test
+    public void testPinningConfig() {
+        Config.getInstance().init(context);
+        ServerConfig serverConfig = Config.getInstance().getServerConfig();
+        OkHttpClient client = OkHttpClientProvider.getInstance().lookup(serverConfig);
+        assertThat(client.certificatePinner().getPins())
+                .contains(new CertificatePinner.Pin(serverConfig.getHost(),
+                        "sha256/9hNxmEFgLKGJXqgp61hyb8yIyiT9u0vgDZh4y8TmY/M="))
+                .contains(new CertificatePinner.Pin(serverConfig.getHost(),
+                        "sha1/2vB3hhEJ98C5efhhWpxtD2wxYek="));
     }
+
+    @Test
+    public void testBuildStepWithCustomPin() throws InterruptedException {
+        ServerConfig serverConfig = ServerConfig.builder()
+                .context(context)
+                .url("https://api.ipify.org")
+                .buildStep(builder -> builder.certificatePinner(
+                        new CertificatePinner.Builder().add("api.ipify.org", "sha1/2vB3hhEJ98C5efhhWpxtD2wxYek=" ).build()))
+                .build();
+
+        OkHttpClient client = OkHttpClientProvider.getInstance().lookup(serverConfig);
+
+        Request request = new Request.Builder()
+                .get()
+                .url("https://api.ipify.org?format=json")
+                .build();
+
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+
+        final boolean[] result = new boolean[1];
+
```

Comment:
@witrisna  can we have a result[0] = false;

Reply:
The default value for boolean primitive type is false, any specific reason we need to set the value?